kola/tests/crio/crio.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crio
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/net/context"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/lang/worker"
+	"github.com/coreos/mantle/platform"
+)
+
+// simplifiedCrioInfo represents the results from crio info
+type simplifiedCrioInfo struct {
+	StorageDriver string `json:"storage_driver"`
+	StorageRoot   string `json:"storage_root"`
+	CgroupDriver  string `json:"cgroup_driver"`
+}
+
+// init runs when the package is imported and takes care of registering tests
+func init() {
+	register.Register(&register.Test{
+		Run:         crioBaseTests,
+		ClusterSize: 1,
+		Name:        `crio.base`,
+	})
+	register.Register(&register.Test{
+		Run:         crioNetwork,
+		ClusterSize: 2,
+		Name:        "crio.network",
+	})
+
+}
+
+// genContainer makes a container out of binaries on the host. This function uses podman
+func genContainer(c cluster.TestCluster, m platform.Machine, name string, binnames []string) {
+	cmd := `tmpdir=$(mktemp -d); cd $tmpdir; echo -e "FROM scratch\nCOPY . /" > Dockerfile;
+	        b=$(which %s); libs=$(sudo ldd $b | grep -o /lib'[^ ]*' | sort -u);
+	        sudo rsync -av --relative --copy-links $b $libs ./;
+	        sudo podman build -t %s .`
+
+	c.MustSSH(m, fmt.Sprintf(cmd, strings.Join(binnames, " "), name))
+}
+
+// crioBaseTests executes multiple tests under the "base" name
+func crioBaseTests(c cluster.TestCluster) {
+	c.Run("version", func(c cluster.TestCluster) {
```

Comment:
I'm not sure we want to commit to having a test for a hard-coded non-OS version specific version test. Tests are typically ran off of master and that would make us have a failing test whenever we want to test builds that have a CRI-O version bump.

Reply:
Fair enough. I'll drop it.